tools/tests/form-control/xhr.test.js

```diff
@@ -0,0 +1,54 @@
+const template = `<div class="form-wrapper">
+<gameface-form-control id="xhr-form" action="http://localhost:${SERVER_PORT}/user" method="get">
+<input name="username" value="name" type="text" />
+<input name="password" value="pass" type="password" />
+<button class="form-element" type="submit">Submit</button>
+</gameface-form-control>
+</div>`;
+
+describe('Form control XHR property tests', () => {
+    beforeAll(() => {
+        const el = document.createElement('div');
+        el.className = 'test-wrapper';
+        el.innerHTML = template;
+
+        const currentEl = document.querySelector('.test-wrapper');
+        if (currentEl) {
+            currentEl.parentElement.removeChild(currentEl);
+        }
+
+        document.body.appendChild(el);
+    });
+
+    afterAll(() => {
+        const currentEl = document.querySelector('.test-wrapper');
+
+        if (currentEl) {
+            currentEl.parentElement.removeChild(currentEl);
+        }
+    });
+
+    it('Should have xhr property exposed', () => {
+        const form = document.querySelector('gameface-form-control');
+        const xhr = form.xhr;
+
+        assert.exists(xhr, 'The xhr property exists.');
+    });
+
+    it('Should be able to attach listeners to the xhr property', async () => {
+        const form = document.querySelector('gameface-form-control');
+        const xhr = form.xhr;
+        const submitButton = document.querySelector('[type="submit"]');
+        this.load = 0;
+        xhr.addEventListener("load", () => this.load += 1);
+
+        click(submitButton);
+        // click twice to make sure the xhr is correctly reset between requests
+        // we create new xhr after the request has finished, this means that the event handler must be called once
+        click(submitButton);

```

Comment:
The submit is asynchronous and this might be flaky.
After the first click the "server" can return a response before the second click - do you think this is possible and what will happen in that case?
Can you elaborate more on the test approach?

Reply:
I tried follow the same pattern we use for the main form-control tests - click submit, wait for server response, check response. I misunderstood the essential of the waitForServerResponse function.
To answer your question - what would happen if the check is done before the response of the second click - if there is a bug we won't catch it.

In order to fix this we need to wait for the first request to complete, subscribe for load of the second and do the checks there - Fixed using the done callback.